backend/headless/backend.c

```diff
@@ -71,10 +70,7 @@ static void backend_destroy(struct wlr_backend *wlr_backend) {
 	wlr_signal_emit_safe(&wlr_backend->events.destroy, backend);
 
 	free(backend->format);
-	if (backend->egl == &backend->priv_egl) {
-		wlr_renderer_destroy(backend->renderer);
-		wlr_egl_finish(&backend->priv_egl);
-	}
+	wlr_renderer_destroy(backend->renderer);
```

Comment:
The headless backend is a little special and can be created with a parent renderer. If that's the case, we shouldn't destroy the renderer here, as it's owned by the parent backend.

We could probably add a `bool has_parent_renderer` field to the backend, and skip destroying the renderer if it's true.

Reply:
Missed that, I just blindly removed both of the `wlr_egl`. Thanks